scipy/stats/_distn_infrastructure.py

```diff
@@ -3041,7 +3044,7 @@ def fun(x):
         return np.power(x, n) * self._pmf(x, *args)
 
     _a, _b = self._get_support(*args)
-    return _expect(fun, _a, _b, self.ppf(0.5, *args), self.inc)
+    return _expect(fun, _a, _b, self._ppf(0.5, *args), self.inc)
```

Comment:
Here and below, `ppf` was being used after input validation was already done. This unnecessarily repeats input validation - and runs the arguments through `_parse_args` again, which is a problem for `_poisson_binom` - so we use the private method instead.